ambry-network/src/main/java/com/github/ambry/network/http2/Http2NetworkClient.java

```diff
@@ -91,7 +91,11 @@ public List<ResponseInfo> sendAndPoll(List<RequestInfo> requestsToSend, Set<Inte
           // We need stream reset to cancel the stream in transmitting.
           RequestInfo requestInfo = streamChannel.attr(Http2NetworkClient.REQUEST_INFO).get();
           if (requestInfo != null) {
-            readyResponseInfos.add(new ResponseInfo(requestInfo, NetworkClientErrorCode.TimeoutError, null));
+            NetworkClientErrorCode errorCode = NetworkClientErrorCode.TimeoutError;
```

Comment:
Not getting a response from server node might also mean network connection issues on the frontend itself. For example, if the connection to the server nodes timeout due to network issues on frontend, it should not mark those nodes with network errors. The server nodes might be totally fine in this case.
Should we actually look into the server code instead, and make sure that in case of disk failures, the server node returns appropriate error message, instead of just timing out?

Reply:
Comments on GetBlobOperation::GetChunk::cleanupExpiredInflightRequests line 801 & PutOperation::PutChunk::cleanupExpiredInflightRequests, also seem indicate something similar.

Reply:
the server code does return correct error code, it's disk_unavailable. And frontend does mark disks as unavailable. 

The problem i am getting in this case, is that in server, the metrics shows there is no incoming requests after disk failures and in frontend, the log shows all the requests going to this particular server time out. But since we are not treating time out as network error, frontend keeps sending requests to this server and keep getting time outs and keep dropping requests. 

Treating timeout as network isn't an ideal solution here, but the problem is that we are only getting timeout from this failing host and that should say something about the health of this server host.

Reply:
Once the code has been verified, are we plan to replace the NetworkClientErrorCode.TimeoutError with NetworkClientErrorCode.NetworkError? 
Since based on below logic, it will treat NetworkError as ReplicaEventType.Node_Timeout anyway.
private void onNetworkEvent(ReplicaId replicaId, NetworkClientErrorCode errorCode) {
    switch (errorCode) {
      case NetworkError:
        clusterMap.onReplicaEvent(replicaId, ReplicaEventType.Node_Timeout);
        break;
      default:
        break;
    }

Reply:
that's right, we will treat timeout as network error and mark this data node as down and stop sending requests to this data node.

One thing I think worth mentioning is that, NetworkErrorCode.Timeout is added with http2 network client. In SocketNetworkClient, there are only ConnectionUnavailable and NetworkError. ConnectionUnavailable means there are too many outgoing requests happening at the same time and we can't check out any socket connection anymore. And NetworkError means as long as there is an error in the socket connection. Timeout is one of those errors. In SocketNetworkClient, we treat timeout (dropped requests) as network error already. 

I suppose the reason why we are not doing that in Http2NetworkClient is that we don't have a way to close the stream like the way we close the socket connection in SocketNetworkClient. But I will try this and see how things go.